### ISSUE TITLE
Test directory incorrectly removed

### ISSUE THREAD

**Issue by psychomario** [snare/scruffy#15]
The following two tests:

 - [`test_environment_full`](https://github.com/snare/scruffy/blob/9a993434e85a8140668a9e884190e714b4e088e2/tests/env_tests.py#L44)
 - [`test_directory`](https://github.com/snare/scruffy/blob/master/tests/file_tests.py#L95)

Fail if the directories `/tmp/scruffy_string_dir` and `/tmp/scruffy_test`(respectively) already exist and are owned by another user. They are left as artifacts after the testing has completed so if the testing is run again as another user it will fail the second time.

Is this expected behaviour or should they be cleaned up? Based on [L103](https://github.com/snare/scruffy/blob/master/tests/file_tests.py#L103) it looks like they should be cleaned up but are not.

I am seeing this behaviour on NixOS with Python27 and Python33.

**Comment by snare** [comment]
Yeah the tests should probably clean these up. I'll look into it.
